### PR TITLE
[Merged by Bors] - chore: reduce mathlib dependencies in Data.Nat.Defs

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
It seems that most of this file belongs in Std.
This changes some proofs to avoid dependencies on mathlib, notably on the typeclass-mediated order lemmas.

---

Maybe you'll like this more :)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by YaelDillies** [comment]
Yes! I like this more. Are there any imports you can remove as a result?

**Comment by Ruben-VandeVelde** [comment]
Just `Tactic.Use` - most of the file could easily move, but there are (relatively few) genuine references to `Nontrivial`, `Set`, and `Function.Injective/Surjective`, as well as `gcongr` annotations and `push_neg` uses.

File: Mathlib/Data/Nat/Defs.lean [hunk 1]
```diff
@@ -1396,7 +1409,7 @@ instance decidableLoHi (lo hi : ℕ) (P : ℕ → Prop) [H : DecidablePred P] :
 instance decidableLoHiLe (lo hi : ℕ) (P : ℕ → Prop) [DecidablePred P] :
     Decidable (∀ x, lo ≤ x → x ≤ hi → P x) :=
   decidable_of_iff (∀ x, lo ≤ x → x < hi + 1 → P x) <|
-    ball_congr fun _ _ => imp_congr Nat.lt_succ_iff Iff.rfl
+    forall_congr' fun _ ↦ forall_congr' (fun _ => imp_congr Nat.lt_succ_iff Iff.rfl)
```
Comment (YaelDillies) [inline Mathlib/Data/Nat/Defs.lean:1412]:
```suggestion
    forall₂_congr fun _ _ ↦ imp_congr Nat.lt_succ_iff Iff.rfl
```

**Review by YaelDillies — COMMENTED** [review]
I think there is little point avoiding `induction'` since it might make it to Std

**Comment by fpvandoorn** [comment]
LGTM (I like the `induction'` changes)

bors merge

**Pull request closed without merging**